src/main/java/graphql/schema/DataFetcherFactory.java

```diff
@@ -0,0 +1,36 @@
+package graphql.schema;
+
+import graphql.PublicSpi;
+
+/**
+ * A DataFetcherFactory allows a level of indirection in providing {@link graphql.schema.DataFetcher}s for graphql fields.
+ *
+ * For example if you are using an IoC container such as Spring or Guice, you can use this indirection to give you
+ * per request late binding of a data fetcher with its dependencies injected in.
+ *
+ * @param <T> the type of DataFetcher
+ */
+@PublicSpi
+public interface DataFetcherFactory<T> {
+
+    /**
+     * Returns a {@link graphql.schema.DataFetcher}
+     *
+     * @param fieldDefinition the {@link graphql.schema.GraphQLFieldDefinition} that needs the data fetcher
+     *
+     * @return a data fetcher
+     */
+    DataFetcher<T> get(GraphQLFieldDefinition fieldDefinition);
```

Comment:
I think it is safer for the future provide an `....Environment` Object containing the `fieldDefitinion`, but also the parent "container".

Reply:
Great point - on it